### DISCUSSION TITLE
500 connect econnrefused 127.0.0.1:3002

### DISCUSSION THREAD

**Discussion by klonb69** [immich-app/immich#2187] [General]
### The bug

After upadate app to 1.52.1 I can't use this in H.A version 2023.40. code error ,,500 connect econnrefused 127.0.0.1' 

### The OS that Immich Server is running on

Home Assistant

### Version of Immich Server

v9.5

### Version of Immich Mobile App

x

### Platform with the issue

- [X] Server
- [ ] Web
- [ ] Mobile

### Your docker-compose.yml content

```YAML
x
```

### Your .env content

```Shell
x
```

### Reproduction steps

```bash
1.x
2.x
3.x
...
x
```

### Additional information

x

**Comment by bo0tzz** [comment]
I've converted this to a discussion as it's very unlikely to be a bug in Immich.

You're running a strange setup, and have provided essentially zero information. You'll need to give a lot more detail for us to be able to help you.

**Reply by klonb69** [reply]
Python v3.10.10
Home Assistant 2023.4.2
Supervisor 2023.04.0
Operating System 9.5
User Interface 20230406.1 - latest

[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/oauth/config, POST} route +3ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/oauth/callback, POST} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/oauth/link, POST} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/oauth/unlink, POST} route +4ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RoutesResolver] SearchController {/search}: +3ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/search, GET} route +3ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/search/config, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/search/explore, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RoutesResolver] ServerInfoController {/server-info}: +1ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/server-info, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/server-info/ping, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/server-info/version, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/server-info/stats, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RoutesResolver] ShareController {/share}: +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/share, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/share/me, GET} route +3ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/share/:id, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/share/:id, DELETE} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/share/:id, PATCH} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RoutesResolver] SystemConfigController {/system-config}: +1ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/system-config, GET} route +3ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/system-config/defaults, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/system-config, PUT} route +1ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/system-config/storage-template-options, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RoutesResolver] UserController {/user}: +1ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/user, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/user/info/:userId, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/user/me, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/user, POST} route +1ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/user/count, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/user/:userId, DELETE} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/user/:userId/restore, POST} route +1ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/user, PUT} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/user/profile-image, POST} route +3ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/user/profile-image/:userId, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RoutesResolver] AssetController {/asset}: +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset/upload, POST} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset/download/:assetId, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset/download-files, POST} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset/download-library, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset/file/:assetId, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset/thumbnail/:assetId, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset/curated-objects, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset/curated-locations, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset/search-terms, GET} route +1ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset/search, POST} route +3ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset/count-by-time-bucket, POST} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset/count-by-user-id, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset, GET} route +1ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset/time-bucket, POST} route +3ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset/:deviceId, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset/assetById/:assetId, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset/:assetId, PUT} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset, DELETE} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset/check, POST} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset/exist, POST} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset/shared-link, POST} route +3ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset/shared-link/add, PATCH} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/asset/shared-link/remove, PATCH} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RoutesResolver] TagController {/tag}: +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/tag, POST} route +3ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/tag, GET} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/tag/:id, GET} route +3ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/tag/:id, PATCH} route +4ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/tag/:id, DELETE} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RoutesResolver] AlbumController {/album}: +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/album/count-by-user-id, GET} route +3ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/album, POST} route +3ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/album/:albumId/users, PUT} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/album/:albumId/assets, PUT} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/album/:albumId, GET} route +3ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/album/:albumId/assets, DELETE} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/album/:albumId, DELETE} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/album/:albumId/user/:userId, DELETE} route +3ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/album/:albumId, PATCH} route +2ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/album/:albumId/download, GET} route +3ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [RouterExplorer] Mapped {/album/create-shared-link, POST} route +3ms
[Nest] 1669245  - 04/11/2023, 10:25:54 AM     LOG [SearchService] Running bootstrap
Request #1681201554943: Request to Node 0 failed due to "ECONNREFUSED connect ECONNREFUSED 127.0.0.1:8108"
Request #1681201554943: Sleeping for 4s and then retrying request...
Request #1681201554943: Request to Node 0 failed due to "ECONNREFUSED connect ECONNREFUSED 127.0.0.1:8108"
Request #1681201554943: Sleeping for 4s and then retrying request...
Request #1681201554943: Request to Node 0 failed due to "ECONNREFUSED connect ECONNREFUSED 127.0.0.1:8108"
Request #1681201554943: Sleeping for 4s and then retrying request...
Request #1681201554943: Request to Node 0 failed due to "ECONNREFUSED connect ECONNREFUSED 127.0.0.1:8108"
Request #1681201554943: Sleeping for 4s and then retrying request...
Request #1681201554943: Request to Node 0 failed due to "ECONNREFUSED connect ECONNREFUSED 127.0.0.1:8108"
Request #1681201554943: Sleeping for 4s and then retrying request...
Request #1681201554943: Request to Node 0 failed due to "ECONNREFUSED connect ECONNREFUSED 127.0.0.1:8108"
Request #1681201554943: Sleeping for 4s and then retrying request...
Request #1681201554943: Request to Node 0 failed due to "ECONNREFUSED connect ECONNREFUSED 127.0.0.1:8108"
Request #1681201554943: Sleeping for 4s and then retrying request...
Request #1681201554943: Request to Node 0 failed due to "ECONNREFUSED connect ECONNREFUSED 127.0.0.1:8108"
Request #1681201554943: Sleeping for 4s and then retrying request...
Request #1681201554943: Request to Node 0 failed due to "ECONNREFUSED connect ECONNREFUSED 127.0.0.1:8108"
Request #1681201554943: Sleeping for 4s and then retrying request...
Request #1681201554943: Request to Node 0 failed due to "ECONNREFUSED connect ECONNREFUSED 127.0.0.1:8108"
Request #1681201554943: Sleeping for 4s and then retrying request...
Request #1681201554943: Request to Node 0 failed due to "ECONNREFUSED connect ECONNREFUSED 127.0.0.1:8108"
Request #1681201554943: Sleeping for 4s and then retrying request...

**Reply by bo0tzz** [reply]
You'll need to check with the people that maintain the installation method you used. It seems they failed to set up Typesense properly?

**Reply by alextran1502** [reply]
What is your CPU, it looks like your system cannot start Typesense

**Reply by klonb69** [reply]
Raspberry PI 4B -- 64-b procesor ARM-8 Cortex-A72 menory 4gb.

**Reply by bo0tzz** [reply]
Are you running any other applications on there besides Immich and HASS? If so, 4gb is pretty tight.

**Reply by klonb69** [reply]
I have a connected drive SSD for usage app. - 450GB (4 GB it's integrated memory RAM  LPDDR4).

**Reply by bo0tzz** [reply]
I meant that 4gb of RAM is pretty tight if you're running more than just Immich.

**Reply by klonb69** [reply]
But this application was already working on this system and the app stopped working :(

**Reply by bo0tzz** [reply]
You're running an unsupported deployment method that we don't know anything about, so you'll need to ask the people who made that package for help.

**Comment by Majika007** [comment]
I have this error also. I speek english poor, sorry...
500  connect ECONNREFUSED 172.20.0.7:3001

My log (end):
[Nest] 1  - 04/08/2023, 4:06:53 PM     LOG [RoutesResolver] AlbumController {/album}: +1ms
[Nest] 1  - 04/08/2023, 4:06:53 PM     LOG [RouterExplorer] Mapped {/album/count-by-user-id, GET} route +1ms
[Nest] 1  - 04/08/2023, 4:06:53 PM     LOG [RouterExplorer] Mapped {/album, POST} route +1ms
[Nest] 1  - 04/08/2023, 4:06:53 PM     LOG [RouterExplorer] Mapped {/album/:albumId/users, PUT} route +2ms
[Nest] 1  - 04/08/2023, 4:06:53 PM     LOG [RouterExplorer] Mapped {/album/:albumId/assets, PUT} route +1ms
[Nest] 1  - 04/08/2023, 4:06:53 PM     LOG [RouterExplorer] Mapped {/album/:albumId, GET} route +1ms
[Nest] 1  - 04/08/2023, 4:06:53 PM     LOG [RouterExplorer] Mapped {/album/:albumId/assets, DELETE} route +2ms
[Nest] 1  - 04/08/2023, 4:06:53 PM     LOG [RouterExplorer] Mapped {/album/:albumId, DELETE} route +1ms
[Nest] 1  - 04/08/2023, 4:06:53 PM     LOG [RouterExplorer] Mapped {/album/:albumId/user/:userId, DELETE} route +1ms
[Nest] 1  - 04/08/2023, 4:06:53 PM     LOG [RouterExplorer] Mapped {/album/:albumId, PATCH} route +1ms
[Nest] 1  - 04/08/2023, 4:06:53 PM     LOG [RouterExplorer] Mapped {/album/:albumId/download, GET} route +2ms
[Nest] 1  - 04/08/2023, 4:06:53 PM     LOG [RouterExplorer] Mapped {/album/create-shared-link, POST} route +1ms
[Nest] 1  - 04/08/2023, 4:06:53 PM     LOG [SearchService] Running bootstrap
Request #1680970013476: Request to Node 0 failed due to "ENOTFOUND getaddrinfo ENOTFOUND typesense"
Request #1680970013476: Sleeping for 4s and then retrying request...
Request #1680970013476: Request to Node 0 failed due to "ENOTFOUND getaddrinfo ENOTFOUND typesense"
Request #1680970013476: Sleeping for 4s and then retrying request...

NODE_VERSION = 16.15.1
ENV YARN_VERSION = 1.22.19

My .env <- sample.env from github, change only UploadLocation.
docker-compose.yml also the sample from github.

My system is OpenMediaVault 6.3.7-5 (Shaitan), and Docker..

My mobil app write error 502.

You can help me?

**Reply by alextran1502** [reply]
Typesense was still trying to come up. Can you bring down the whole stack and bring it up again?

**Reply by alextran1502** [reply]
What is server URL you are putting in for the mobile app? Can you help taking a screenshot?

**Reply by Majika007** [reply]
![immich1](https://user-images.githubusercontent.com/97151954/230755843-3f9be0a6-0896-4261-b270-795f347066e9.png)
![scr01](https://user-images.githubusercontent.com/97151954/230756146-cec72db7-a5bf-472a-9fce-2d38a8ec100e.jpg)
![scr02](https://user-images.githubusercontent.com/97151954/230756147-e1d66a0e-5400-40c0-a7ca-007806edbca1.jpg)
